registration.php

```diff
@@ -14,518 +35,621 @@
 
             // Query for Residential address details
-            $empResUpdate = "UPDATE Address
-          SET street = '$resStreet', city = '$resCity', zip = '$resZip', state = '$resState'
-          WHERE empId = '$employeeIdUpdate' 
-          AND addressType = 'residence' ";
-
-            $result = mysqli_query($conn, $empUpdate);
-            $ofcResult = mysqli_query($conn, $empOfcUpdate);
-            $resResult = mysqli_query($conn, $empResUpdate);
+            $empResData = ['street' => $resStreet,
+                'city' => $resCity,
+                'zip' => $resZip,
+                'state' => $resState];
+            $condition = ['column' => 'empId',
+                'operator' => '=',
+                'val' => "'$employeeIdUpdate' AND addressType = 'residence'"];
+            $obj->insertUpdate('Address', $empResData, $condition, TRUE);
 
-            if (!$result && !$ofcResult && !$resResult)
-            {
-                echo 'Update failed' . mysql_error();
-            }
+            //Redirect User to Employee List Page
+            header('Location:list.php');
         }
         else
         {
-            // Insert personal details
-            $employeeInsert = "INSERT INTO Employee(title, firstName, middleName, lastName, 
-          dateOfBirth, gender, phone, email, maritalStatus, empStatus, commId, image, note, employer)
-          VALUES ('$title', '$firstName', '$middleName', '$lastName', '$dob', '$gender', $phone, 
-          '$email', '$marStatus', '$empStatus', '$communication','$name',
-          '$note', '$employer')";
-
-            $result = mysqli_query($conn, $employeeInsert);
+            array_push($empData, ['password' => $password]);
 
-            // Id of the last inserted record
-            $employeeId = mysqli_insert_id($conn);
+            // Insert data   
+            $employeeId = $obj->insertUpdate('Employee', $empData);
 
-            // Query to insert employee details
-            $address = "INSERT INTO Address(addressType, street, city, zip, state, empId) 
-            values('office', '$ofcStreet', '$ofcCity', '$ofcZip', '$ofcState', '$employeeId'), 
-            ('residence', '$resStreet', '$resCity', '$resZip', '$resState', '$employeeId')";
+            // Insert office address detail
+            $empOfcData = array('addressType' => 'office', 'street' => $ofcStreet,
+                'city' => $ofcCity, 'zip' => $ofcZip, 'state' => $ofcState, 'empId' => $employeeId);
+            $obj->insertUpdate('Address', $empOfcData);
 
-            $addressResult = mysqli_query($conn, $address);
-
-            if (!$result && !$addressResult)
-            {
-                echo 'Insertion failed' . mysql_error();
-            }
+            // Insert residence address detail
+            $empResData = array('addressType' => 'residence', 'street' => $resStreet,
+                'city' => $resCity, 'zip' => $resZip, 'state' => $resState, 'empId' => $employeeId);
```

Comment:
Here $employeeId is passed to insert address information for employee but not checked in previous query whether really employee is inserted successfully or not.